lib/tesla.ex

```diff
@@ -517,19 +521,34 @@ defmodule Tesla do
       iex> Tesla.build_url(url, [page: 2, status: true])
       "http://api.example.com?user=3&page=2&status=true"
 
+      # default encoding `:www_form`
+      iex> Tesla.build_url("http://api.example.com", [user_name: "John Smith"])
+      "http://api.example.com?user_name=John+Smith"
+
+      # specified encoding strategy :rfc3986
+      iex> Tesla.build_url("http://api.example.com", [user_name: "John Smith"], :rfc3986)
+      "http://api.example.com?user_name=John%20Smith"
   """
-  @spec build_url(Tesla.Env.url(), Tesla.Env.query()) :: binary
-  def build_url(url, []), do: url
+  @spec build_url(Tesla.Env.url(), Tesla.Env.query(), :rfc3986 | :www_form) :: binary
```

Comment:
```suggestion
  @type encoding_strategy :: :rfc3986 | :www_form

  @spec build_url(Tesla.Env.url(), Tesla.Env.query(), encoding_strategy) :: binary
```

Also related to this, what happen if some weird encoding is needed? 

As an example: https://github.com/elixir-tesla/tesla/issues/538#issuecomment-1242348413 I had to do that for a project, and, I am not too familiar with whatever RFCs exists out there.

So, what I created was part of some RFC?
Should we allow some callback to handle the encoding as a scape goat? Maybe add it when somebody asks for it? What would be the workload then?